packages/babel/test/as-input-plugin.js

```diff
@@ -0,0 +1,430 @@
+
+import { getCode } from '../../../util/test';
+
+import babelPlugin from '..';
+
+process.chdir(__dirname);
+
+function getLocation(source, charIndex) {
+  const lines = source.split('\n');
+  const len = lines.length;
+
+  let lineStart = 0;
+
+  for (let i = 0; i < len; i += 1) {
+    const line = lines[i];
+    // +1 for newline
+    const lineEnd = lineStart + line.length + 1;
+
+    if (lineEnd > charIndex) {
+      return { line: i + 1, column: charIndex - lineStart };
+    }
+
+    lineStart = lineEnd;
+  }
+
+  throw new Error('Could not determine location of character');
+}
+
+function replaceConsoleLogProperty({ types: t }) {
+  return {
+    name: 'replace-console-log-property',
+    visitor: {
+      MemberExpression(path, state) {
+        const { opts } = state;
+        if (path.node.property.name === 'log') {
+          path.get('property').replaceWith(t.identifier(opts.replace));
+        }
+      }
+    }
+  };
+}
+
+async function generate(input, babelOptions = {}, generateOptions = {}, rollupOptions = {}) {
+  const bundle = await rollup({
+    input,
+    plugins: [babelPlugin({ babelHelpers: 'bundled', ...babelOptions })],
+    ...rollupOptions
+  });
+
+  return getCode(bundle, {
+    format: 'cjs',
+    ...generateOptions
+  });
+}
+
+test('runs code through babel', async (t) => {
+  const code = await generate('fixtures/basic/main.js');
+  t.false(code.includes('const'));
+  t.is(
+    code,
+    `'use strict';
+
+var answer = 42;
+console.log("the answer is ".concat(answer));
+`
+  );
+});
+
+test('adds helpers', async (t) => {
+  const code = await generate('fixtures/class/main.js');
+  t.true(code.includes('function _classCallCheck'));
+});
+
+test('adds helpers in loose mode', async (t) => {
+  const code = await generate('fixtures/class-loose/main.js');
+  t.true(code.includes('function _inherits'));
+});
+
+test('does not babelify excluded code', async (t) => {
+  const code = await generate('fixtures/exclusions/main.js', { exclude: '**/foo.js' });
+  // eslint-disable-next-line no-template-curly-in-string
+  t.false(code.includes('${foo()}'));
+  t.true(code.includes('=> 42'));
+  t.is(
+    code,
+    `'use strict';
+
+const foo = () => 42;
+
+console.log("the answer is ".concat(foo()));
+`
+  );
+});
+
+test('generates sourcemap by default', async (t) => {
+  const bundle = await rollup({
+    input: 'fixtures/class/main.js',
+    plugins: [babelPlugin({ babelHelpers: 'bundled' })]
+  });
+
+  const {
+    output: [{ code, map }]
+  } = await bundle.generate({ format: 'cjs', sourcemap: true });
+
+  const target = 'log';
+  const smc = new SourceMapConsumer(map);
+  const loc = getLocation(code, code.indexOf(target));
+  const original = smc.originalPositionFor(loc);
+
+  t.deepEqual(original, {
+    source: 'fixtures/class/main.js'.split(nodePath.sep).join('/'),
+    line: 3,
+    column: 12,
+    name: target
+  });
+});
+
+test('works with proposal-decorators (#18)', async (t) => {
```

Comment:
I wonder if we should adjust the issue numbers. Maybe `rollup/rollup-plugin-babel#18`? We can also just leave it if you like.